### ISSUE TITLE
liblzma issue with cvmfs py3-v4 on local el7 machines (Trac #2356)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2348]
On Chiba cluster we use Linux grappa 3.10.0-327.4.5.el7.x86_64 which unfortunately comes with xz (XZ Utils) 5.1.2alpha/liblzma 5.1.2alpha. Thus icetray doesn't work with cvmfs py3-v4.

error messages below

```text
ImportError                               Traceback (most recent call last)
<ipython-input-1-4fb3d2024319> in <module>()
----> 1 from icecube import icetray

/misc/disk19/users/lulu/Software/snobo/build_snobo_python3/lib/icecube/icetray/__init__.py in <module>()
      1 from icecube.load_pybindings import load_pybindings
----> 2 load_pybindings(__name__, __path__)
      3 
      4 from icecube.icetray.test_module import I3TestModuleFactory
      5 from icecube.icetray.traysegment import traysegment, traysegment_inherit, module_altconfig

/misc/disk19/users/lulu/Software/snobo/build_snobo_python3/lib/icecube/load_pybindings.py in load_pybindings(name, path)
     65     import imp, sys
     66     thismod = sys.modules[name]
---> 67     m = imp.load_dynamic(name, path[0] + ".so")
     68     sys.modules[name] = thismod # Some python versions overwrite the Python
     69                                 # module entry with the C++ one. We don't want

/cvmfs/icecube.opensciencegrid.org/py3-v4/RHEL_7_x86_64/lib/python3.6/imp.py in load_dynamic(name, path, file)
    341         spec = importlib.machinery.ModuleSpec(
    342             name=name, loader=loader, origin=path)
--> 343         return _load(spec)
    344 
    345 else:

ImportError: /usr/lib64/liblzma.so.5: version `XZ_5.2' not found (required by /cvmfs/icecube.opensciencegrid.org/py3-v4/RHEL_7_x86_64/lib/libarchive.so.13)

```

Is it possible to include liblzma in cvmfs? Thanks!

Migrated from https://code.icecube.wisc.edu/ticket/2356
```json
{
    "status": "closed", 
    "changetime": "2019-09-27T15:17:28", 
    "description": "On Chiba cluster we use Linux grappa 3.10.0-327.4.5.el7.x86_64 which unfortunately comes with xz (XZ Utils) 5.1.2alpha/liblzma 5.1.2alpha. Thus icetray doesn't work with cvmfs py3-v4.\n\nerror messages below\n\n\n{{{\nImportError                               Traceback (most recent call last)\n<ipython-input-1-4fb3d2024319> in <module>()\n----> 1 from icecube import icetray\n\n/misc/disk19/users/lulu/Software/snobo/build_snobo_python3/lib/icecube/icetray/__init__.py in <module>()\n      1 from icecube.load_pybindings import load_pybindings\n----> 2 load_pybindings(__name__, __path__)\n      3 \n      4 from icecube.icetray.test_module import I3TestModuleFactory\n      5 from icecube.icetray.traysegment import traysegment, traysegment_inherit, module_altconfig\n\n/misc/disk19/users/lulu/Software/snobo/build_snobo_python3/lib/icecube/load_pybindings.py in load_pybindings(name, path)\n     65     import imp, sys\n     66     thismod = sys.modules[name]\n---> 67     m = imp.load_dynamic(name, path[0] + \".so\")\n     68     sys.modules[name] = thismod # Some python versions overwrite the Python\n     69                                 # module entry with the C++ one. We don't want\n\n/cvmfs/icecube.opensciencegrid.org/py3-v4/RHEL_7_x86_64/lib/python3.6/imp.py in load_dynamic(name, path, file)\n    341         spec = importlib.machinery.ModuleSpec(\n    342             name=name, loader=loader, origin=path)\n--> 343         return _load(spec)\n    344 \n    345 else:\n\nImportError: /usr/lib64/liblzma.so.5: version `XZ_5.2' not found (required by /cvmfs/icecube.opensciencegrid.org/py3-v4/RHEL_7_x86_64/lib/libarchive.so.13)\n\n}}}\n\n\nIs it possible to include liblzma in cvmfs? Thanks!\n", 
    "reporter": "lulu", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1569597448681537", 
    "component": "cvmfs", 
    "summary": "liblzma issue with cvmfs py3-v4 on local el7 machines", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2019-09-15T16:57:20", 
    "milestone": "", 
    "owner": "david.schultz", 
    "type": "rumor / allegation"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20190916T12:49:21: david.schultz commented:

> `/cvmfs/icecube.opensciencegrid.org/py3-v4/RHEL_7_x86_64/lib/liblzma.so.5.2.4` seems to exist.  So either this is a problem with your environment, or with our cmake detection scripts.
> 
> Maybe find one of the software experts at the meeting (possibly me), and we can debug it with you?

**Comment by icecube-trac** [comment]
Trac update at 20190923T16:27:48: 
* olivas changed status from "new" to "assigned"
* olivas changed owner from "" to "david.schultz"

**Comment by icecube-trac** [comment]
Trac update at 20190923T19:11:02: 
* david.schultz changed type from "defect" to "rumor / allegation"
* david.schultz commented:

> Is this still an issue?

**Comment by icecube-trac** [comment]
Trac update at 20190927T01:52:26: lulu commented:

> Replying to [comment:3 david.schultz]:
> > Is this still an issue?
> The fix in py3-v4.0.1 changing (bin to lib) worked perfectly. Thanks!

**Comment by icecube-trac** [comment]
Trac update at 20190927T15:17:28: 
* olivas changed status from "assigned" to "closed"
* olivas changed resolution from "" to "fixed"
